### PR TITLE
std: unsafe-wrap personality::gcc

### PR DESCRIPTION
**Pull request by workingjubilee**
Nothing seems obviously wrong with these implementations except for some unanswered questions. Admittedly, I don't want to burn excessive time on exceptional exception handlers. Thus this is mostly a brute-force syntactic wrapping and some comments where they seemed correct, creating another largely whitespace diff.

Part of https://github.com/rust-lang/rust/issues/127747

try-job: armhf-gnu

### CODE REVIEW

File: library/std/src/sys/personality/gcc.rs [hunk 1]
```diff
@@ -92,151 +93,162 @@ const UNWIND_DATA_REG: (i32, i32) = (4, 5); // a0, a1
...
-        // ARM EHABI personality routine.
-        // https://web.archive.org/web/20190728160938/https://infocenter.arm.com/help/topic/com.arm.doc.ihi0038b/IHI0038B_ehabi.pdf
-        //
-        // Apple 32-bit ARM (but not watchOS) uses the default routine instead
-        // since it uses SjLj unwinding.
+    if #[cfg(all(
+            target_arch = "arm",
+            not(all(target_vendor = "apple", not(target_os = "watchos"))),
+            not(target_os = "netbsd"),
+        ))] {
```
Comment (workingjubilee) [inline library/std/src/sys/personality/gcc.rs:100]: I found the current cfg condition excruciating to read, thus this rearrangement.

**Comment by workingjubilee** [comment]
Trying armhf-gnu since that seems like the one that'll trip up in the final build if any.

@bors try

File: library/std/src/sys/personality/gcc.rs (hunk 1 above)
Comment (ChrisDenton) [reply library/std/src/sys/personality/gcc.rs:100]: `all(not(all...not(` yeah.

**Comment by ChrisDenton** [comment]
lgtm!

@bors r+

**Pull request merged**